templates/fine-tune-function-calling/README.md

```diff
@@ -0,0 +1,419 @@
+    [92muser: [0mCan you tell me the current stock price of AAPL?
+    [94massistant: [0m[TOOL_CALLS] [ {"name": "get_stock_price", "arguments": '{"symbol": "AAPL"}'} ] [/TOOL_CALLS]
+    [92muser: [0m[TOOL_RESULTS] [{"status": "success", "data": {"symbol": "AAPL", "price": 150.75}}] [/TOOL_RESULTS]
+    [94massistant: [0mThe current stock price of AAPL is $150.75. 
+    [92muser: [0mWhat about the stock price of MSFT?
+    [94massistant: [0m[TOOL_CALLS] [ {"name": "get_stock_price", "arguments": '{"symbol": "MSFT"}'} ] [/TOOL_CALLS]
+    [92muser: [0m[TOOL_RESULTS] [{"status": "success", "data": {"symbol": "MSFT", "price": 295.40}}] [/TOOL_RESULTS]
+    [94massistant: [0mThe current stock price of MSFT is $295.40. 
+    [92muser: [0mThank you for the information.
+    [94massistant: [0mYou're welcome! If you have any other questions, feel free to ask. 
+    
+
+
+Let's make a train, validation and test split and save the datasets in the `jsonl` format.
+
+
+```python
+train_ds, val_ds, test_ds = processed_ds.split_proportionately([0.8, 0.1])
+test_ds, _  = test_ds.split_at_indices([200]) # restrict to 200 examples for testing
+```
+
+
+```python
+# inspect final counts
+train_ds.count(), val_ds.count(), test_ds.count()
+```
+
+
+
+
+    (9036, 1129, 200)
+
+
+
+
+```python
+# save the datasets to jsonl format
+save_to_jsonl(train_ds, "glaiveai-function-calling-v2-train.jsonl")
+save_to_jsonl(val_ds, "glaiveai-function-calling-v2-val.jsonl")
+save_to_jsonl(test_ds, "glaiveai-function-calling-v2-test.jsonl")
+```
+
+# Step 2: Fine-tuning 
+
+For fine-tuning, you have two options with Anyscale:
+1. Fine-tuning on the Anyscale Platform through our fine-tuning template 
+    - This would be the preferred route for those wishing to get more flexibility in choice of models and hyperparameters, better monitoring, etc.
+2. Fine-tuning through Anyscale's serverless endpoints
+    - A quick and easy way to fine-tune a model via an OpenAI compatiable SDK.
+
+For this guide, we will use `Llama-3-8B-Instruct` as the base model for fine-tuning.
+
+
+
+## Step 2(a): Fine-tuning on the Anyscale Platform
+
+Head over to the Anyscale Platform: https://console.anyscale.com/v2 and spin up the "Fine-tune LLMs" template (under "AI application templates")
+
+<p align="center">
+  <img src="./assets/templates.png" alt="Templates">
+</p>
+
+
+
+Follow the instructions to run your fine-tuning job.
+
+## Step 2(b): Fine-tuning through serverless endpoints
+First, obtain your credentials from the [Anyscale platform](https://console.anyscale.com/v2/) and upload the training and validation files in the [fine-tuning tab](https://console.anyscale.com/v2/fine-tuning?fine-tuning-tab=files). Make a note of the file IDs for each. This will be passed to the fine-tuning job.
```

Comment:
They don't need to upload the files separately from the finetuning job. They can upload data at the time of finetuning submision.

Reply:
so instead of going first under files tab, they can just go to jobs tab and create and upload at the same time.

Reply:
So here you are kinda mixing UI and SDK. 

We can just make it SDK only. You can include code for data upload here as well. And just punt on file upload via UI.

Reply:
Ah... I actually missed this part of the SDK! Have updated the upload instructions now.